WeatherStationClient.py

```diff
@@ -0,0 +1,258 @@
+from abc import ABCMeta, abstractmethod
+import asyncio
+import aiohttp
+from Crypto.Hash import HMAC
+from Crypto.Hash import SHA256
+from datetime import datetime
+from dateutil.tz import tzlocal
+
+class UnauthorizedException(Exception):
+    pass
+    
+class BadRequestException(Exception):
+    pass
+    
+class ConflictException(Exception):
+    pass
+
+class ForbiddenException(Exception):
+    pass
+ 
+class ServerErrorException(Exception):
+    def __init__(self, message):
+        self.message = message
+
+class ApiClient:        
+    def __init__(self, auth):
+        self._auth = auth
+    
+    async def _send(self, *args):
+        resp = await self._auth._make_request(*args)
+        return resp
+    
+    async def user_information(self):
+        """Reading user information."""
+        return await self._send('GET', 'user')
+    
+    # Czy wolno sprobowac przetestowac te metode? O ile pamietam mielismy pamietac by nie wywolywac metod ktore moga byc destruktywne?
+    async def update_user_information(self, user_data):
```

Comment:
I am not sure if we have to implement methods other than get.

Reply:
Maybe we could split this big client into few specialized clients. ie. one per section mentioned in documentation: User, System, Station ...

Reply:
Uncertain because it would seem that some sections interoperate heavily, eg docu for the data section mentions retrieving supported disease models from system/diseases.

...

Well I have an idea. Maybe lets keep this big client but instead of hosting all methods lets make it host objects that will host relevant methods? So that the usage would be something like `client.User.get_user()`